src/internal/utils.js

```diff
@@ -48,7 +48,12 @@ export function validate_store(store, name) {
 }
 
 export function subscribe(component, store, callback) {
-	component.$$.on_destroy.push(store.subscribe(callback));
+	let unsub = store.subscribe(callback);
+	// Prevent memory leaks for RxJS users.
+	if (typeof unsub === 'object' && typeof unsub.unsubscribe === 'function') {
```

Comment:
Does the RxJS API mean that there'll always be an `unsubscribe` function? If so would this work?

```suggestion
	if (unsub.unsubscribe) {
```

Reply:
Yes, RxJS guarantees a `Subscription` is always returned from the `subscribe` call, and that will always be an object with an `unsubscribe` method. The extra code I added was mostly because I wasn't sure what your APIs could return there, so I was being defensive.